fix(xychart): align y-axis to start from zero (#6517)

duyaan opened the pull request:
## :bookmark_tabs: Summary

Fixes Y-axis in XY bar charts not starting at zero. Ensures 0 aligns with X-axis baseline.

Resolves #6517

## :straight_ruler: Design Decisions

- Forced Y-axis to always include 0 in range.
- Adjusted layout/render logic to match visual expectation.
-the graph is now accurate 

## :clipboard: Tasks

- [x] Read the contribution guidelines
- [x] Added tests
- [x] Added doc if necessary
- [x] Generated a changeset using `pnpm changeset`
<img width="904" alt="Screenshot 2025-04-23 at 6 54 35 PM" src="https://github.com/user-attachments/assets/51a2a070-35e0-45a6-9369-63207ad76e5d" />

duyaan committed:
fix(xychart): align y-axis to start from zero (#6517)

duyaan committed:
docs(changeset): align y-axis to start from zero

sidharthv96 commented:
Please check [the changes](https://app.argos-ci.com/mermaid/mermaid/builds/4015), there are many breaking changes. 


<img width="926" alt="image" src="https://github.com/user-attachments/assets/2676a9f6-3504-4f25-a66f-7efe9555e605" />

duyaan committed:
fix(xychart): ensure bar chart Y-axis starts from zero without breaking other charts

- Added optional  field to XYChartAxisConfig.
- Modified LinearAxis to conditionally include 0 in the domain if  is true.
- Only bar charts set , ensuring Y-axis starts from 0.
- Other charts (line, area, etc.) remain dynamically scaled without forcing 0.
- Fixes issue #6517.

duyaan commented:
@sidharthv96  hi , 

Thanks for the feedback I have updated the pull request based on the review:

- Added an optional forceZeroYStart field to the XYChartAxisConfig interface.

- LinearAxis now conditionally forces the Y-domain to start from 0 only if forceZeroYStart is set to true.

- The forceZeroYStart option is enabled only for bar charts, ensuring that the 0 baseline is correctly aligned with the X-axis without affecting other chart types 

- Other charts remain unaffected, preserving their dynamic Y-axis scaling.

This should resolve the issue #6517 without introducing breaking changes.

sidharthv96 reviewed (changes requested):
Adding a config is not the right approach. 

![image](https://github.com/user-attachments/assets/7c66fe43-f6c8-45db-9023-94917a989306)

![image](https://github.com/user-attachments/assets/75a18781-a855-4680-815a-e21576a98b9b)

In both these cases, the bars should start from the X Axis, and not be above or below it. Visual changes where that small shift happens is not a breaking change, the problem is the cases like above where the bar continue under the X Axis.
